src/quo2/components/settings/settings_item/view.cljs

```diff
@@ -0,0 +1,118 @@
+    [quo2.components.avatars.user-avatar.view :as user-avatar]
+    [utils.i18n :as i18n]))
+
+(defn status-description
+  [{:keys [online? text]} blur? theme]
+  [rn/view
+   {:style style/status-container}
+   [rn/view
+    {:style (style/status-dot online? blur?)}]
+   [text/text
+    {:size  :paragraph-2
+     :style (style/color blur? theme)}
+    (if online? (i18n/label :t/online-now) text)]])
+
+(defn text-description
+  [description-props blur? theme]
+  [rn/view
+   {:style style/sub-container}
+   [text/text
+    {:size  :paragraph-2
+     :style (style/color blur? theme)}
+    (:text description-props)]
+   (when (:icon description-props)
+     [icon/icon (:icon description-props)
+      (merge (style/color blur? theme)
+             {:size            16
+              :container-style {:margin-left 4}})])])
+
+(defn description-component
+  [description description-props blur? theme]
+  (case description
+    :text           [text-description description-props blur? theme]
+    :text-plus-icon [text-description description-props blur? theme]
+    :status         [status-description description-props blur? theme]
+    nil))
+
+(defn image-component
+  [image image-props description tag blur? theme]
+  [rn/view
+   {:style (style/image-container description tag image)}
+   (case image
+     :icon        [icon/icon image-props (style/color blur? theme)]
+     :avatar      [user-avatar/user-avatar image-props]
+     :icon-avatar [icon-avatar/icon-avatar image-props]
+     nil)])
+
+(defn tag-component
+  [tag tag-props]
+  (case tag
+    :positive [status-tags/status-tag
+               {:status          {:type :positive}
+                :label           (i18n/label :t/positive)
+                :size            :small
+                :container-style {:margin-top 8}}]
+    :context  [context-tag/view
+               (merge tag-props
+                      {:type            :icon
+                       :size            24
+                       :container-style {:margin-top 8
+                                         :align-self :flex-start}})]
+    nil))
+
+(defn label-component
+  [label label-props blur? theme]
+  [rn/view {:accessibility-label :label-component}
+   (case label
+     :text    [text/text
+               {:style (style/color blur? theme)}
+               label-props]
+     :color   [rn/view
+               {:style (style/label-dot label-props)}]
+     :preview [preview-list/view {:type (:type label-props)} (:data label-props)]
+     nil)])
+
+(defn action-component
+  [action action-props blur? theme]
+  [rn/view {:style {:margin-left 12}}
+   (case action
+     :arrow    [icon/icon :i/chevron-right (style/color blur? theme)]
+     :button   [button/button
+                {:type     :outline
+                 :size     24
+                 :on-press (:on-press action-props)}
+                (:button-text action-props)]
+     :selector [selectors/toggle action-props]
+     nil)])
+
+(defn- internal-view
```

Comment:
perhaps use `:as props` and pass it along and destructure as you need.
```
(defn- internal-view
  [{:keys [title image image-props description description-props action action-props label label-props
           tag tag-props on-press in-card? blur? accessibility-label theme] :as props}]
           
  [rn/pressable
   {:style               (style/container in-card? tag)
    :on-press            on-press
    :accessibility-label accessibility-label}
   [rn/view {:style style/sub-container}
    [image-component props]
    [rn/view {:style {:margin-left 12}}
     [text/text {:weight :medium} title]
     [description-component props]
     [tag-component props]]]
   [rn/view {:style style/sub-container}
    [label-component props]
 ```

Reply:
@J-Son89 fixed 👍